Use more r-value references in ElementRareData

nt1m opened the pull request:
#### 28fb95152b231d4fc81b202389cc8776ca9c9476
<pre>
Use more r-value references in ElementRareData
<a href="https://bugs.webkit.org/show_bug.cgi?id=252184">https://bugs.webkit.org/show_bug.cgi?id=252184</a>
rdar://105412403

Reviewed by Chris Dumez.

Add more consistency with other methods, which also call WTFMove.

* Source/WebCore/dom/ElementRareData.h:
(WebCore::ElementRareData::setAttributeMap):
(WebCore::ElementRareData::setComputedStyle):
(WebCore::ElementRareData::setClassList):
(WebCore::ElementRareData::setDataset):
(WebCore::ElementRareData::setPartList):

Canonical link: <a href="https://commits.webkit.org/260224@main">https://commits.webkit.org/260224@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7d7d86de8d04723111bb07953d514d46c88e4e66

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107571 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16628 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40472 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116728 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116143 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111466 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18053 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/7942 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99738 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113330 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13628 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96820 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41289 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95537 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28446 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83081 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9625 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29799 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10290 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/6696 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15782 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49380 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7065 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11852 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
unique_ptr is not copyable so there isn't really any risk of people copying them by mistake. That said, I like to use r-value references whenever possible and being consistent, so r=me.

nt1m committed:
Use more r-value references in ElementRareData
https://bugs.webkit.org/show_bug.cgi?id=252184
rdar://105412403

Reviewed by Chris Dumez.

Add more consistency with other methods, which also call WTFMove.

* Source/WebCore/dom/ElementRareData.h:
(WebCore::ElementRareData::setAttributeMap):
(WebCore::ElementRareData::setComputedStyle):
(WebCore::ElementRareData::setClassList):
(WebCore::ElementRareData::setDataset):
(WebCore::ElementRareData::setPartList):

Canonical link: https://commits.webkit.org/260224@main
